Report/Template.Rmd

```diff
@@ -243,17 +251,262 @@ datasets = read.and.clean(dataset = "easySHARE_rel6_0_0.rda", wav = 99)
+Functions that calculate group shares per country and labor participation rates can subsequently be applied onto the defined grid. 
+
+Besides the target data frame, `group.share` takes a boolean variable as an input that specifies whether aggregation on a national level is intended. The share of males or females (per country) is calculated by finding the sum of occurences and dividing by their total, represented by the length of the vector. A similar procedure that allows for multiple subsets and based upon `dplyr` is used in `labor`.
+
+```{r, eval = FALSE}
+# Function for calculation group shares per country
+group.share = function(df, y, agg.country = TRUE) {
+    if (agg.country) {
+        final = tapply(df[[y]], df$country, function(x) {
+            val = sum(x)/length(x)
+            return(val)
+        })
+    } else { # return total if country-level aggregation not specified
+        final = sum(df[[y]])/length(df[[y]])
+    }
+    return(final)
+}
+
+# Function for calculating labor participation rate per country / gender / age
+labor = function(df, .country, .gender, .age) {
+    dataset = df %>% 
+        filter(country == .country & gender == .gender & get(.age) == TRUE)
+    denom = nrow(dataset)
+    numer = with(dataset, sum(labor_ft | labor_pt))
+    perc  = round(numer/denom, 4)
+    
+    return(perc)
+}
+
+```
+
+Within the code below, `args.list` contains three sub-lists divided by countriy (`c`), gender (`g`) and age (`a`). The list is passed as an argument to `labor`, which are simulatenously iterated over the sub-lists using `pmap`. This function has an advantage over a similar functions, since it allows multiple arguments and uses parallelization by default.  
+Including `with` to specify the data frame on which we expand the grid upon, modifies the data frame in a new environment. This simplifies name assignment. 
+
+```{r, eval = FALSE}
+
+# use expand.grid to get all combinations of country / gender / age
+agegroups = names(df.out[, 4:6])
+args      = with(df.out, 
+                 expand.grid(levels(country), levels(gender), agegroups, 
+                             stringsAsFactors = FALSE))
+args.list = list(c = as.vector(t(args[1])), 
+                 g = as.vector(t(args[2])), 
+                 a = as.vector(t(args[3])))
+
+# Map over multiple arguments
+output   = pmap(args.list, function(c, g, a) labor(df.out, c, g, a))
+output.v = do.call("c", output)
+
+labor.part.share    = cbind(args, output.v)
+labor.part.share.df = spread(labor.part.share, Var3, output.v) %>% 
+    arrange(desc(Var2)) %>% 
+    set_colnames(c("Country", "Gender", 
+                   paste0("Age ", rep(c("50-54", "55-59", "60-64")))))
+```
+
+The resulting `output` list can be easily bound into a data frame and arranged using `dplyr`. From this package we also use `spread` in order to subset by gender. Using the pipe operator `%>%` keeps our code compact and avoids unnecessary variable assignment and inefficient memory requirements.
```

Comment:
`spread` is a `tidyr` function and it converts a long data.frame to a wide data.frame. In our case we spread the variable `Var3` (which is the age groups) into 3 separate columns. I would say don't worry about it too much... maybe focus on the idea from above of using the two functions `group.share` and `labor` to create the summary tables, then we do a bunch of weird transformations to get them into the format we want, then concatenate the totals to the bottom of the tables.

Reply:
halfway integrated. Will check the code again and then add some sentences.